specs/governance/governor.md

```diff
@@ -0,0 +1,702 @@
+    - [`ProposalExecuted`](#proposalexecuted)
+    - [`ProposalCreated`](#proposalcreated)
+    - [`ProposalDeadlineUpdated`](#proposaldeadlineupdated)
+    - [`ProposalThresholdSet`](#proposalthresholdset)
+    - [`VotingDelaySet`](#votingdelayset)
+    - [`VotingPeriodSet`](#votingperiodset)
+    - [`QuorumNumeratorUpdated`](#quorumnumeratorupdated)
+- [Storage](#storage)
+- [Types](#types)
+  - [`ProposalVote`](#proposalvote)
+  - [`ProposalCore`](#proposalcore)
+- [Proposal Types](#proposal-types)
+- [Modules](#modules)
+- [Security Considerations](#security-considerations)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+The `Governor` contract implements the core governance logic for creating, voting, and executing proposals.
+This contract uses the `GovernanceToken` contract for voting power snapshots, and the `ProposalTypesConfigurator`
+for proposal types.
+
+## Constants
+
+| Constant          | Value                           | Description |
+| ----------------- | ------------------------------- | ----------- |
+| `BALLOT_TYPEHASH` |  `0x150214d74d59b7d1e90c73fc22ef3d991dd0a76b046543d4d80ab92d2a50328f` | The EIP-712 typehash for the ballot struct  |
+| `EXTENDED_BALLOT_TYPEHASH` |  `0x7f08F3095530B67CdF8466B7a923607944136Df0` | The EIP-712 typehash for the extended ballot struct |
+| `COUNTING_MODE` | `support=bravo&quorum=against,for,abstain&params=modules` | The configuration for supported values for `castVote` and how votes are counted |
+| `PERCENT_DIVISOR` |  `10000` | The maximum value of `quorum` and `approvalThreshold` for proposal types  |
+| `PROPOSAL_TYPES_CONFIGURATOR` |  `0x67ecA7B65Baf0342CE7fBf0AA15921524414C09f` | The address of the proposal types configurator contract  |
+| `VERSION` |  `2` | The version of the `Governor` contract  |
+| `TOKEN` |  `0x4200000000000000000000000000000000000042` | The address of the `GovernanceToken` contract  |
+
+## Interface
+
+### Core Functions
+
+#### `cancel`
+
+Cancels a proposal. A proposal MUST only be cancellable by the manager and ONLY while it's in an active state,
+i.e. not executed or cancelled already.
+
+```solidity
+function cancel(address[] memory _targets, uint256[] memory _values, bytes[] memory _calldatas, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `cancelWithModule`
+
+Similar to `cancel`, but for proposals that were created with a module.
+
+```solidity
+function cancelWithModule(address _module, bytes memory _proposalData, bytes32 _descriptionHash) external returns (uint256)
+```
+
+This function MUST emit the `ProposalCanceled` event and return the ID of the proposal that was cancelled.
+
+#### `castVote`
+
+Cast a vote on a proposal.
+
+```solidity
+function castVote(uint256 _proposalId, uint8 _support) external returns (uint256)
```

Comment:
What is `uint8 _support`?

Reply:
specified in https://github.com/ethereum-optimism/specs/pull/315/commits/b737496a8c35fc389d6205672ed1601899a157a4